docs/modules/record-encryption/con-record-encryption-overview.adoc

```diff
@@ -73,7 +73,11 @@ It is recommended to use the KMS in a high availability (HA) configuration.
 Key rotation involves periodically replacing cryptographic keys with new ones and is considered a best practice in cryptography.
 
 The filter allows the rotation of Key Encryption Keys (KEKs) within the Key Management System (KMS). 
-When a KEK is rotated, the new key material is used for newly produced records. Existing records, encrypted with older KEK versions, remain decryptable as long as the previous KEK versions are still in the KMS.
+When a KEK is rotated, the new key material is eventually used for newly produced records. Existing records, encrypted with older KEK versions, remain decryptable as long as the previous KEK versions are still in the KMS.
+
+When the KEK is rotated in the external KMS, it will take up to 2 hours (by default) before all records produced by the filter
+contain a DEK encrypted with the new key material. This is because existing encrypted DEKs are used for a configurable
+amount of time after creation, the Filter caches the encrypted DEK.
```

Comment:
Should we also document the nuclear option of  a (rolling) restart of instances if people are responding to a key compromise?

Reply:
done